qiskit/aqua/operators/converters/circuit_sampler.py

```diff
@@ -243,7 +250,12 @@ def sample_circuits(self,
 
         Returns:
             The dictionary mapping ids of the CircuitStateFns to their replacement StateFns.
+        Raises:
+            AquaError: if extracted circuits are empty.
         """
+        if not circuit_sfns and not self._transpiled_circ_cache:
+            raise AquaError('CircuitStateFn is empty and there is no cache.')
+
         if circuit_sfns or not self._transpiled_circ_cache:
```

Comment:
I am trying to understand the logic here. So is no circuit_sfns are passed (this is default) and we have no cache then its an error. So far seems fine. So we can have one (circuit_sfns), or the other (cache), or both potentially and proceed.

Now the next test is if we have circuit_sfns or no cache. Clearly for it to register on the 'or' part of the if there would be no circuit_sfns. But no circuit sfns and no cache we rejected above. And clearly when you look inside the if the code immediately chooses to to do list comprehension over circuit_sfns. Maybe the or part of the if is not useful - its certainly confusing to me trying to read the code.